misc/distributed_search/outer.sh

```diff
@@ -0,0 +1,238 @@
+first_search_config_filename="config.yaml"
+second_search_config_filename="config.yaml"
+# Search flags
+predict_library=1
+first_search=1
+mbr_library=1
+second_search=1
+lfq=1
+
+while [[ "$#" -gt 0 ]]; do
+	case $1 in
+		# Search parameters
+		--input_directory) input_directory="$2"; shift ;;
+		--input_filename) input_filename="$2"; shift ;;
+		--target_directory) target_directory="$2"; shift ;;
+		--library_path) library_path="$2"; shift ;;
+		--fasta_path) fasta_path="$2"; shift ;;
+		--first_search_config_filename) first_search_config_filename="$2"; shift ;;
+		--second_search_config_filename) second_search_config_filename="$2"; shift ;;
+		# SLURM parameters
+		--nnodes) nnodes="$2"; shift ;;
+		--ntasks_per_node) ntasks_per_node="$2"; shift ;;
+		--cpus) cpus="$2"; shift ;;
+		--mem) mem="$2"; shift ;;
+		# Search flags: if a flag is present, set to true
+		--predict_library) predict_library="$2"; shift ;;
+		--first_search) first_search="$2"; shift ;;
+		--mbr_library) mbr_library="$2"; shift ;;
+		--second_search) second_search="$2"; shift ;;
+		--lfq) lfq="$2"; shift ;;
+		*) echo "Unknown parameter $1"; shift ;;
+	esac
+	shift
+done
+
+	# create logs directory if it does not exist
+	mkdir -p ./logs
+
+	# remove and create target directory
+	rm -rf ${target_directory}
+	mkdir ${target_directory}
+
+	predicted_library_directory="${target_directory}predicted_speclib/"
+	mkdir -p ${predicted_library_directory}
+
+	first_search_directory="${target_directory}first_search/"
+	mkdir -p ${first_search_directory}
+
+	mbr_library_directory="${target_directory}mbr_library/"
+	mkdir -p ${mbr_library_directory}
+	
+	mbr_progress_directory="${target_directory}mbr_library/chunk_0/.progress/"
+	mkdir -p ${mbr_progress_directory}
+
+	second_search_directory="${target_directory}second_search/"
+	mkdir -p ${second_search_directory}
+
+	lfq_directory="${target_directory}lfq/"
+	mkdir -p ${lfq_directory}
+
+	lfq_progress_directory="${target_directory}lfq/chunk_0/.progress/"
+	mkdir -p ${lfq_progress_directory}
+
+### PREDICT LIBRARY ###
+
+if [[ "$predict_library" -eq 1 ]]; then
+
+	# generate config without rawfiles and with fasta
+	python ./speclib_config.py \
+	--input_directory "${input_directory}" \
+	--target_directory "${target_directory}" \
+	--fasta_path "${fasta_path}" \
+	--library_path "${library_path}" \
+	--config_filename "${first_search_config_filename}" \
+
+	# log current directory and navigate to predicted speclib directory
+	home_directory=$(pwd)
+	cd "${predicted_library_directory}"	
+
+	# call alphadia to predict spectral library
+	echo "Predicting spectral library with AlphaDIA"
+	sbatch --array=${slurm_array} \
+	--wait --nodes=1 \
+	--ntasks-per-node=${ntasks_per_node} \
+	--cpus-per-task=${cpus} \
+	--mem=${mem} \
+	--output=./../../logs/%j-speclib-slurm.out \
+	--export=ALL --wrap="alphadia --config=speclib_config.yaml"
```

Comment:
Yes, this was done because the speclib prediction config should be the same as the first_search config, so I set it to simple be copied and hard-coded in the predict_speclib folder